ros2topic/ros2topic/verb/pub.py

```diff
@@ -66,6 +66,8 @@ def publisher(message_type, topic_name, values):
     module = importlib.import_module(package_name + '.msg')
     msg_module = getattr(module, message_name)
     values_dictionary = yaml.load(values)
+    if not isinstance(values_dictionary, dict):
+        return 'The passed value needs to be a dictionary in YAML format'
```

Comment:
is it appropriate to return the error message here instead of raise?

Reply:
Because the return value of this function is also the return value of `main`.

A few lines below (line 82) it does the same.